program/src/processor.rs

```diff
@@ -37,16 +37,19 @@ pub fn process_instruction<'a>(
 fn get_stake_activating_and_deactivating(accounts: &[AccountInfo]) -> ProgramResult {
     let accounts_iter = &mut accounts.iter();
     let stake_info = next_account_info(accounts_iter)?;
-    if *stake_info.owner != stake::program::id() {
-        return Err(ProgramError::IllegalOwner);
-    }
-
     let stake_history_info = next_account_info(accounts_iter)?;
     if *stake_history_info.key != sysvar::stake_history::id() {
         return Err(ProgramError::InvalidArgument);
     }
 
     let mut stake_view = GetStakeActivatingAndDeactivatingReturnData::default();
+
+    // if it's not a SOL stake, that's fine, all zeros
+    if *stake_info.owner != stake::program::id() {
+        set_return_data(bytemuck::bytes_of(&stake_view));
```

Comment:
Do you think it makes sense to add a byte discriminator to the return data to indicate whether the account was valid/found or not? It does not make a difference for Paladin, but if the program has other uses in the future might be interesting to know.

Reply:
People can check the ownership on the calling side if they're very sensitive to it. I wouldn't worry about it to be honest